widgetssdk/src/main/java/com/glia/widgets/messagecenter/MessageCenterActivity.kt

```diff
@@ -23,11 +64,39 @@ class MessageCenterActivity : AppCompatActivity(),
         val messageCenterView = binding.messageCenterView
         messageCenterView.onFinishListener = this
         messageCenterView.onNavigateToMessagingListener = this
+        messageCenterView.onAttachFileListener = this
 
-        val controller = Dependencies.getControllerFactory().getMessageCenterController(configuration?.queueId)
         messageCenterView.setController(controller)
     }
 
+    override fun selectAttachmentFile(type: String) {
+        getContent.launch(type)
+    }
+
+    override fun takePhoto() {
+        if (ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.CAMERA
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            getPermission.launch(Manifest.permission.CAMERA)
+            return
+        }
```

Comment:
It doesn't work with this permission.

Reply:
Maybe you misunderstood me, I meant that no need for camera permission to launch `getImage.launch(uri)` `ActivityResultLauncher`, because this way we are getting result Uri from the native camera app without using `CAMERA` ourselves.

Reply:
"If you have <uses-permission android:name="android.permission.CAMERA" /> declared in your manifest, you need to have permission before launching, otherwise java.lang.SecurityException: Permission Denial"
https://stackoverflow.com/a/65526167

Reply:
I also was surprised, but according to the answer from StackOverflow, it works this way.

Reply:
According to this discussion https://stackoverflow.com/a/33290921/8472205 that's for API version <= 23.
By the way, I checked, and we can request to take a picture without CAMERA permission for API >= 24, our min SDK version is 24.

Reply:
Without permission, I got an exception on my Samsung with API version 33.
```
java.lang.SecurityException: Permission Denial: starting Intent { act=android.media.action.IMAGE_CAPTURE flg=0x3 cmp=com.sec.android.app.camera/.Camera clip={text/uri-list hasLabel(0) {U(content)}} (has extras) } from ProcessRecord{742812b 25967:com.glia.exampleapp.debug/u0a474} (pid=25967, uid=10474) with revoked permission android.permission.CAMERA
```

Reply:
I think it would be best to add some comment in the code for future reference

Reply:
@AndriiHorishniiMOC I think I found the reason why you're receiving that exception. According to [this](https://stackoverflow.com/a/59187440/8472205) If you are using the Camera APIs(declared `CAMERA` permission in `Manifest`) in addition to using `ACTION_IMAGE_CAPTURE`, you must request the permission at runtime.
I tested it on a clean project, so I didn't receive this exception.

Reply:
@DavDo This is exactly what I tried to explain.

Reply:
Just to clarify, we cannot remove `<uses-permission android:name="android.permission.CAMERA" />` from `AndroidManifest` because we have places that use Camera not by calling `getImage.launch(uri)`, right?

Reply:
@AndriiHorishniiMOC I don't know why I didn't realize it before.
@andrews-moc Right, for example for video calls.

Reply:
@andrews-moc We have video calls, so I guess we can't remove the camera permission.